filament/backend/src/metal/MetalState.h

```diff
@@ -169,6 +171,15 @@ class StateCache {
 
     void setDevice(id<MTLDevice> device) noexcept { mDevice = device; }
 
+    void removeIf(utils::Invocable<bool(const StateType&)> fn) noexcept {
+        for (auto it = mStateCache.begin(); it != mStateCache.end(); it++) {
+            const auto& [key, _] = *it;
+            if (UTILS_UNLIKELY(fn(key))) {
+                mStateCache.erase(it);
```

Comment:
Is this safe?  Iterating through the cache while removing key-value pairs?

Reply:
I'm glad you mentioned that, because I double checked and turns out `tsl::robin_map` treats iterators differently than `std::unordered_map` (`robin_map` invalidates _all_ iterators upon `.erase()`). So I need to figure something else out.

Reply:
I wonder if you could use `utils/Bimap.h` instead.

```
 * A semi-generic custom bimap. This bimap stores a key/value pair and can retrieve
 * the value from the key and the key from the value.
 * It is optimized for large keys and small values.  The keys are stored out-of-line and are
 * never moved.
```

I don't know if it fits your usecase?

Reply:
Not quite. I need to be able to match all keys that fit a given criteria.